### PR TITLE
Project init

### PR DESCRIPTION
**Pull request by henry2004y**
This PR aims at clarifying the initialization step for projects. `initialize()` is required for projects because extra parameters may be read for a specific project which are then used to set the internal parameters; otherwise all the initialization should be done in the class constructors. In order to reflect the sequences, `initialize()` is moved after `addParameters()` and `setParameters()`. `initialize()` is enforced for the inherited classes by setting `virtual void initialize() = 0` in the base class. Similarly, `setProjectBField()` is enforced for the inherited classes and renamed to `setInitialBField` to reflect its purpose.

### CODE REVIEW

File: projects/project.cpp [hunk 1]
```diff
@@ -600,85 +545,61 @@ namespace projects {
...
+   } else if (Parameters::projectName == "") {
+      cerr << "No project specified! Please set 'project' parameter!" << endl;
+      MPI_Abort(MPI_COMM_WORLD, 1);
+   } else {
       cerr << "Unknown project name!" << endl;
-      abort();
+      MPI_Abort(MPI_COMM_WORLD, 1);
    }
 
-   getObjectWrapper().project = rvalue;
```
Comment (ykempf) [inline projects/project.cpp:681]: Why did you remove this? Is it unused for sure?

**Comment by ykempf** [comment]
>  In order to reflect the sequences, `initialize()` is moved after `addParameters()` and `setParameters()`.

I don't see where the order changed, can you point that out to me? Maybe I'm just not fit for this today... ;)

File: projects/project.cpp (hunk 1 above)
Comment (henry2004y) [reply projects/project.cpp:681]: It is set twice, once here and once in Vlasiator.cpp

**Comment by henry2004y** [comment]
> > In order to reflect the sequences, `initialize()` is moved after `addParameters()` and `setParameters()`.
> 
> I don't see where the order changed, can you point that out to me? Maybe I'm just not fit for this today... ;)

Ah, I mean the order of definition, not the order of execution. Sorry about the confusion.

File: projects/project.cpp (hunk 1 above)
Comment (ykempf) [reply projects/project.cpp:681]: All right.
